suplemon/modules/application_state.py

```diff
@@ -50,7 +58,11 @@ def restore_states(self):
         for file in self.app.get_files():
             path = file.get_path()
             if path in self.storage.get_data().keys():
-                self.set_file_state(file, self.storage[path])
+                state = self.storage[path]
+                if "hash" not in state:
+                    self.set_file_state(file, state)
+                elif state["hash"] == self.get_hash(file.get_editor()):
+                    self.set_file_state(file, state)
```

Comment:
What do we expect to happen in the `else` case?

Reply:
In the else case, if I figured this out correctly nothing should be done. We don't want to touch the initial state at all if the file has changed since the last time suplemon modified it. The if itself is intended for backwards compatibility.

Reply:
Cool, we can simplify these 2 lines then via `.get()` which is `[]` but with a `None` fallback:

```python
# DEV: We use `.get()` as `hash` could not exist yet
if state.get("hash") == self.get_hash(file.get_editor()):
  self.set_file_state(file, state)
```

Reply:
How smooth is that! I didn't remember dicts had `get`.

Reply:
OTOH does this still solve the backward compatibility part? The idea is to restore the state always if a user's state file doesn't contain a hash for a particular file. If the hash is there the restore should only be done if the hash matches the current file hash.

Reply:
I don't think I follow. Is there a time when `self.get_hash()` would return `None` as well?

Reply:
Let's see if I can clarify my intentions:
1. Since the hash is a new feature, it might not exist in any of a users states that have previously been stored.
2. When a user upgrades to the next version, none of their previous file states will be restored because they're missing the hash (not what I want). Hashes will ofcourse be added to files they open and then close in the new version.
3. If a saved state has been stored before this new version I want that state to be restored regardless of the new hash feature. Otherwise we'll be ignoring all the previously stored states.
4. If a hash does exist only then we use it for verifying whether we should restore a particular file state.

Reply:
Ah, okay. It's less about the code inline here but more the overall implementation